internal/qbittorrent/sync_manager.go

```diff
@@ -654,6 +779,115 @@ func (sm *SyncManager) primaryTrackerDomain(torrent qbt.Torrent) string {
 	return ""
 }
 
+func trackerMessageMatches(message string, patterns []string) bool {
+	text := strings.TrimSpace(strings.ToLower(message))
+	if text == "" {
+		return false
+	}
+
+	for _, pattern := range patterns {
+		if strings.Contains(text, pattern) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func statusFiltersRequireTrackerData(statuses []string) bool {
+	for _, status := range statuses {
+		switch status {
+		case "unregistered", "tracker_down":
+			return true
+		}
+	}
+
+	return false
+}
+
+func (sm *SyncManager) torrentIsUnregistered(torrent qbt.Torrent) bool {
+	for _, tracker := range torrent.Trackers {
+		if tracker.Status != qbt.TrackerStatusNotWorking {
+			continue
+		}
+
+		if trackerMessageMatches(tracker.Message, defaultUnregisteredStatuses) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (sm *SyncManager) torrentTrackerIsDown(torrent qbt.Torrent) bool {
+	for _, tracker := range torrent.Trackers {
+		if tracker.Status != qbt.TrackerStatusNotWorking {
```

Comment:
Reverse the logic?

If any tracker has a working status, return false, since the torrent has a working tracker.

Reply:
Good call 👍